Update release description (#1)
awalach-org/service-catalog.wiki-clone: All-About-Releases.md

A release is a [GitHub Release](https://github.com/kubernetes-incubator/service-catalog/releases), its associated Git tag, and a set of [Docker images on quay.io](https://quay.io/organization/kubernetes-service-catalog) that are all tagged with the Git tag. The images never change, and are intended to be an immutable artifact. Anyone should be able to reference them with confidence that they will never change.

# Details

Release pipeline is very similar to the the pipeline that is triggered for every commit on master branch, but it sets different versions and creates Helm chart:

![Service Catalog Release Process](assets/service-catalog-release-process.svg)

> As shown on the graph we support two major versions of Service Catalog:

Reviewer: ```suggestion
We support two major versions of Service Catalog:
```

> - v0.2 - deprecated, API-server based

Reviewer: ```suggestion
- v0.2 - deprecated, API-server-based
```

> - v0.3 - current master, CRDs based

Reviewer: ```suggestion
- v0.3 - the current master branch, CRDs-based
```

> Release process creates three types of artifacts:

Reviewer: Is that so? 🤔

> - docker images

Reviewer: ```suggestion
- Docker images
```

- svcat binaries

> - helm charts

Reviewer: ```suggestion
- Helm charts
```

They are created for the following platforms: amd64, arm, arm64, ppc64le, s390x

> All artifacts from v0.2 branch that are created after creation of version `v0.3` are appended with `-v0.2` suffix. It means we can easily create bug-fix releases for the deprecated version without interfering with current releases.

Reviewer: ```suggestion
All artifacts from the v0.2 branch that are created after the `v0.3` version, are followed with the `-v0.2` suffix. It means that we can easily create bug-fix releases for the deprecated version without interfering with the current releases.
```
